src/app/webserver/app-pools/app-pool.component.ts

```diff
@@ -1,48 +1,76 @@
-import {Component, OnInit, Inject} from '@angular/core';
-import {ActivatedRoute, Router} from '@angular/router';
-import {DiffUtil} from '../../utils/diff';
-import {OptionsService} from '../../main/options.service';
-import {ApplicationPool} from './app-pool';
-import {AppPoolsService} from './app-pools.service';
-import { BreadcrumbsService } from 'header/breadcrumbs.service';
-import { BreadcrumbsRoot, AppPoolsCrumb, Breadcrumb } from 'header/breadcrumb';
-import { AppPoolsModuleName } from 'main/settings';
+import { Component, OnInit, Inject } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DiffUtil } from '../../utils/diff';
+import { ApplicationPool } from './app-pool';
+import { AppPoolsService } from './app-pools.service';
+import { AppPoolsModuleName, AppPoolModuleIcon } from 'main/settings';
+import { BreadcrumbsResolver, FeatureContext } from 'common/feature-vtabs.component';
+import { Breadcrumb, BreadcrumbsRoot, AppPoolsCrumb } from 'header/breadcrumb';
+import { ModelStatusUpdater, UpdateType } from 'header/model-header.component';
+import { TitlesService } from 'header/titles.service';
+import { resolveAppPoolRoute } from 'webserver/webserver-routing.module';
+
+const crumbsRoot = BreadcrumbsRoot.concat(AppPoolsCrumb);
+class AppPoolBreadcrumbResolver implements BreadcrumbsResolver {
+    resolve(model: FeatureContext): Breadcrumb[] {
+        const pool = <ApplicationPool> model;
+        return crumbsRoot.concat(<Breadcrumb>{ label: pool.name, routerLink: [resolveAppPoolRoute(pool.id)] });
+    }
+}
+
+class AppPoolStatusUpdater extends ModelStatusUpdater {
+    constructor(
+        service: AppPoolsService,
+        pool: ApplicationPool,
+    ) {
+        super(
+            AppPoolsModuleName,
+            AppPoolModuleIcon,
+            pool.name,
+            pool,
+            new Map<UpdateType, () => void>([
+                [UpdateType.Recycle, () => service.recycle(pool)],
+                [UpdateType.Start, () => service.start(pool)],
+                [UpdateType.Stop, () => service.stop(pool)],
+                // [UpdateType.delete, () => service.delete(pool)],
```

Comment:
Please remove this line.

Reply:
I wasn't sure of delete button is appropriate to put here, but it can be implemented here. So I am leaving it here as a note